python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -805,12 +816,18 @@ def pack_value(
         * frozenset
     """
     descent_path = _root(val) if descent_path is None else descent_path
-    return _pack_value(val, whitelist_map=whitelist_map, descent_path=descent_path)
+    return _prepare_value(
+        val,
```

Comment:
is `pack_value` above a public API? Is that we are not changing the name of it? It's a bit confusing at first plush not that it is not actually packing a value, but instead wrapping it in the wrapper thing in prep for iterative serailization.

Reply:
Yea `pack_value` is a public API for cases when a structure of primitives is needed without going directly to json string. Often this is used to as an intermediate before serializing to json but nesting in some other structure. 

To support both `pack_value` and `serialize_value` , their shared traversal function `_prepare_value` takes an `object_handler` function to either go all the way to primitives in one shot [1] or just wrap for incremental serialization [2]